### PR TITLE
Uniswap Volume KPI

### PR DESCRIPTION
**Pull request by cc7768**
Updated version of #389

### CODE REVIEW

File: Implementations/uniswap-volume-kpi.md [hunk 1]
```diff
@@ -0,0 +1,540 @@
...
+## Summary
+
+This calculation is intended to track the volume (denominated in USD) across both V2 and V3 Uniswap on Ethereum.
+
+The recommended method is to query TheGraph for Uniswap's volume. This document will detail the method that will be used to query TheGraph and also mitigate some attack vectors to manipulate the volume. The main vector of attack that we are concerned about is wash-trading where people create new pools based on worthless tokens and then drive volume up by trading the token back and forth with themself -- In practice, it would be difficult to make this profitable given the amount of trade volume that already occurs and the benefits/costs of doing so (the gas fees alone would likely be higher than the increased payoff one would receive).
+
+However, we think it's important to protect the KPI from this and similar attack vectors and so we will compute the volume in two steps:
+
+1. Identify a set of pairs/pools.
+  - We choose to use the criteria that Uniswap itself uses to only track "meaningful usage".
```
Comment (smb2796) [inline Implementations/uniswap-volume-kpi.md:13]: We also picked pools that met this criteria at a specific timestamp. Can you include that here?

File: Implementations/uniswap-volume-kpi.md [hunk 2]
```diff
@@ -0,0 +1,540 @@
...
+2. Compute the volume
+  - In order to further fend off manipulation, we will use an average over 30 days rather than using a single day's volume.
```
Comment (smb2796) [inline Implementations/uniswap-volume-kpi.md:15]: 30 day average daily volume, correct?

File: Implementations/uniswap-volume-kpi.md [hunk 3]
```diff
@@ -0,0 +1,540 @@
...
+We include a description of our methodology so that it could be replicated, but, we also include a script that could be run to generate the values.
+
+
+### Uniswap v2
+
+**Pair criteria for inclusion**
+
+Uniswap v2 has the following criteria for inclusion in their volume metric:
+
+1. It cannot be listed in the [explicitly blacklisted tokens](https://github.com/Uniswap/uniswap-v2-subgraph/blob/master/src/mappings/helpers.ts#L22)
```
Comment (smb2796) [inline Implementations/uniswap-volume-kpi.md:38]: Can you add the snapshot timestamp here as well and in the UNIv3 pool inclusion?

File: Implementations/uniswap-volume-kpi.md [hunk 4]
```diff
@@ -0,0 +1,540 @@
+## Uniswap Volume KPI Calculation
+
+
```
Comment (mhairimca) [inline Implementations/uniswap-volume-kpi.md:3]: Can this not use the generic KPI UMIP?

Comment (smb2796) [reply Implementations/uniswap-volume-kpi.md:3]: This is using the generic KPI UMIP

Comment (smb2796) [reply Implementations/uniswap-volume-kpi.md:3]: Things being merged into `Implementations` are not UMIPs, but rather using ancillary data

**Review by smb2796 — APPROVED** [review]

**Pull request merged**